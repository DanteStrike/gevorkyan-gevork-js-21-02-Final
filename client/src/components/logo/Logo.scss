@import "src/assets/variables";

.logo {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 155px;
  height: 50px;
  padding: 8px 0;
  color: var(--primary-font-color);

  &:hover {
    color: inherit;
  }

  &__icon {
    font-size: 30px;
  }

  &__title {
    font-family: var(--secondary-font-family);
    font-size: 24px;
    font-weight: 700;
  }
}

@media (max-width: $tabletWidth) {
  .logo {
    width: 130px;
    height: 40px;

    &__icon {
      font-size: 24px;
    }

    &__title {
      font-size: 20px;
    }
  }
}
