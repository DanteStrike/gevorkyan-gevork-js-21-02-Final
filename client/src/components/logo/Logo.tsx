import React from 'react';
import './Logo.scss';
import {TwitterCircleFilled} from '@ant-design/icons';
import CustomLink from '../custom-link/CustomLink';
import {RoutePath} from '../../enums';

function Logo() {
  return (
    <CustomLink to={RoutePath.BASE}>
      <div className="logo">
        <TwitterCircleFilled className="logo__icon" />
        <p className="logo__title">Delta World</p>
      </div>
    </CustomLink>
  );
}

export default React.memo(Logo);
