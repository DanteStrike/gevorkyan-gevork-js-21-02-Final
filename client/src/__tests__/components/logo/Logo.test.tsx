import {shallow} from 'enzyme';
import React from 'react';
import Logo from '../../../components/logo/Logo';

describe(`Component Logo should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
});
