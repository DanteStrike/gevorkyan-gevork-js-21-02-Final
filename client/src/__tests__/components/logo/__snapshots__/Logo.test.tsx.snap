// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Logo should render correctly snapshot default test 1`] = `
<Memo(CustomLink)
  to="/"
>
  <div
    className="logo"
  >
    <ForwardRef(TwitterCircleFilled)
      className="logo__icon"
    />
    <p
      className="logo__title"
    >
      Delta World
    </p>
  </div>
</Memo(CustomLink)>
`;
